adr_core_local_impl/src/adr_repo/mod.rs

```diff
@@ -256,91 +239,78 @@ pub fn split_path<'a>(base_path: &'a Path, file_path: &'a Path) -> (&'a Path, &'
+    transition_to(TransitionStatus::CANCELLED, base_path, file_path, "")
 }
 
-pub fn transition_to(transition: TransitionStatus, base_path: &Path, from: &str, by: &str) -> io::Result<bool> {
-    let adr_from = match build_adr(base_path, Path::new(from)){
+pub fn transition_to(transition: TransitionStatus, base_path: &Path, from_path: &str, by_path: &str) -> io::Result<bool> {
+    let origin = match build_adr(base_path, Path::new(from_path)){
         Ok(adr) => adr,
         Err(why) => {
-            error!(get_logger(), "Got error [{:?}] while getting ADR [{}]", why, from);
+            error!(get_logger(), "Got error [{:?}] while getting ADR [{}]", why, from_path);
             panic!();
         },
     };
 
-    let updated_adr_from_tuple = adr_from.update_status(transition);
+    let mut new_adr = origin.clone();
+    let update_status = new_adr.update_status(transition);
 
     //if transition has been declined, we can stop here
-    match updated_adr_from_tuple.1 {
+    match update_status {
         true => {
-            debug!(get_logger(), "ADR [{}] has a new status [{}]", updated_adr_from_tuple.0.path().as_str(), updated_adr_from_tuple.0.status.as_str());
-            match by.is_empty() {
+            let transition_adr = |adr: &Adr, path: &str| -> io::Result<bool> {
+                let content = &adr.content;
+                fs::write(path, content)?;    
+                Ok(update_status)
+            };
+            debug!(get_logger(), "ADR [{}] has a new status [{}]", new_adr.path().as_str(), new_adr.status.as_str());
+            match by_path.is_empty() {
                 true => {
-                    let content = &updated_adr_from_tuple.0.content;
-                    fs::write(from, content)?;
-        
+                    let return_value = transition_adr(&new_adr, from_path);
                     info!(get_logger(), 
                         "Transitioned [{}] from [{}] to [{}]", 
-                        updated_adr_from_tuple.0.path().as_str(), adr_from.status.as_str(), updated_adr_from_tuple.0.status.as_str());
-        
-                    Ok(updated_adr_from_tuple.1)
+                        origin.path().as_str(), origin.status.as_str(), new_adr.status.as_str());
+                    return_value
                 }
                 false => {
-                    let adr_by = build_adr(base_path, Path::new(by))?;
-                    let updated_adr_by_tuple = adr_by.update_status(TransitionStatus::revert(transition));
-
-                    //if status has been accepted
-                    match updated_adr_by_tuple.1 {
-                        true => {
-                            let updated_adr_from = updated_adr_from_tuple.0.add_reference(format!("{}", updated_adr_by_tuple.0.file_name).as_str());
-                            let updated_adr_by = updated_adr_by_tuple.0.add_reference(format!("{}", updated_adr_from.file_name).as_str());
```

Comment:
found the issue, I forgot to add these `add_reference()` calls back in... :disappointed_relieved:

Reply:
making a commit to fix this now.